cpp/tests/copying/concatenate_tests.cpp

```diff
@@ -228,6 +212,17 @@ TEST_F(StringColumnTest, ConcatenateTooManyColumns)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
 }
 
+TEST_F(StringColumnTest, ConcatenateTooLarge)
+{
+  std::string big_str(1000000, 'a');
+  cudf::test::strings_column_wrapper input{big_str, big_str, big_str, big_str, big_str};
+  std::vector<cudf::column_view> input_cols;
+  for (int i = 0; i < 500; ++i) {
```

Comment:
could you please add a comment about the limit we're crossing over in this test, or derive the iteration count from `std::numeric_limits<size_type>()`?